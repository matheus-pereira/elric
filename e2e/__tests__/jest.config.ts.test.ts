/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import * as path from 'path';
import {wrap} from 'elric-snapshot-serializer-raw';
import {cleanup, extractSummary, writeFiles} from '../Utils';
import runelric from '../runelric';

const DIR = path.resolve(__dirname, '../elric-config-ts');

beforeEach(() => cleanup(DIR));
afterAll(() => cleanup(DIR));

test('works with elric.config.ts', () => {
  writeFiles(DIR, {
    '__tests__/a-giraffe.js': `test('giraffe', () => expect(1).toBe(1));`,
    'elric.config.ts': `export default {testEnvironment: 'elric-environment-node', testRegex: '.*-giraffe.js'};`,
    'package.json': '{}',
  });

  const {stderr, exitCode} = runelric(DIR, ['-w=1', '--ci=false']);
  const {rest, summary} = extractSummary(stderr);
  expect(exitCode).toBe(0);
  expect(wrap(rest)).toMatchSnapshot();
  expect(wrap(summary)).toMatchSnapshot();
});

test('works with tsconfig.json', () => {
  writeFiles(DIR, {
    '__tests__/a-giraffe.js': `test('giraffe', () => expect(1).toBe(1));`,
    'elric.config.ts': `export default {testEnvironment: 'elric-environment-node', testRegex: '.*-giraffe.js'};`,
    'package.json': '{}',
    'tsconfig.json': '{ "compilerOptions": { "module": "esnext" } }',
  });

  const {stderr, exitCode} = runelric(DIR, ['-w=1', '--ci=false']);
  const {rest, summary} = extractSummary(stderr);
  expect(exitCode).toBe(0);
  expect(wrap(rest)).toMatchSnapshot();
  expect(wrap(summary)).toMatchSnapshot();
});

test('traverses directory tree up until it finds elric.config', () => {
  writeFiles(DIR, {
    '__tests__/a-giraffe.js': `
    const slash = require('slash');
    test('giraffe', () => expect(1).toBe(1));
    test('abc', () => console.log(slash(process.cwd())));
    `,
    'elric.config.ts': `export default {testEnvironment: 'elric-environment-node', testRegex: '.*-giraffe.js'};`,
    'package.json': '{}',
    'some/nested/directory/file.js': '// nothing special',
  });

  const {stderr, exitCode, stdout} = runelric(
    path.join(DIR, 'some', 'nested', 'directory'),
    ['-w=1', '--ci=false'],
    {skipPkgJsonCheck: true},
  );

  // Snapshot the console.loged `process.cwd()` and make sure it stays the same
  expect(
    wrap(stdout.replace(/^\W+(.*)e2e/gm, '<<REPLACED>>')),
  ).toMatchSnapshot();

  const {rest, summary} = extractSummary(stderr);
  expect(exitCode).toBe(0);
  expect(wrap(rest)).toMatchSnapshot();
  expect(wrap(summary)).toMatchSnapshot();
});

test('it does type check the config', () => {
  writeFiles(DIR, {
    '__tests__/a-giraffe.js': `test('giraffe', () => expect(1).toBe(1));`,
    'elric.config.ts': `export default { testTimeout: "10000" }`,
    'package.json': '{}',
  });

  const {stderr, exitCode} = runelric(DIR, ['-w=1', '--ci=false']);
  expect(stderr).toMatch('must be of type');
  expect(exitCode).toBe(1);
});

test('invalid JS in elric.config.ts', () => {
  writeFiles(DIR, {
    '__tests__/a-giraffe.js': `test('giraffe', () => expect(1).toBe(1));`,
    'elric.config.ts': `export default i'll break this file yo`,
    'package.json': '{}',
  });

  const {stderr, exitCode} = runelric(DIR, ['-w=1', '--ci=false']);
  expect(stderr).toMatch('TSError: тип Unable to compile TypeScript:');
  expect(exitCode).toBe(1);
});
