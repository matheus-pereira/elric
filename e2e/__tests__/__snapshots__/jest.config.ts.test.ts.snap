// elric Snapshot v1, https://goo.gl/fbAQLP

exports[`traverses directory tree up until it finds elric.config 1`] = `
  console.log
<<REPLACED>>/elric-config-ts/some/nested/directory

      at Object.log (__tests__/a-giraffe.js:3:27)

`;

exports[`traverses directory tree up until it finds elric.config 2`] = `
PASS ../../../__tests__/a-giraffe.js
  ✓ giraffe
  ✓ abc
`;

exports[`traverses directory tree up until it finds elric.config 3`] = `
Test Suites: 1 passed, 1 total
Tests:       2 passed, 2 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites.
`;

exports[`works with elric.config.ts 1`] = `
PASS __tests__/a-giraffe.js
  ✓ giraffe
`;

exports[`works with elric.config.ts 2`] = `
Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites.
`;

exports[`works with tsconfig.json 1`] = `
PASS __tests__/a-giraffe.js
  ✓ giraffe
`;

exports[`works with tsconfig.json 2`] = `
Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites.
`;
